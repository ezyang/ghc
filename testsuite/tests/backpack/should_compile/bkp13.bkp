package h where
    signature H(T) where
        data T
package p where
    include h
    module B(T(..)) where
        data T = T
    signature H(T(..), f) where
        import B(T(..))
        f :: a -> a
    module A(T) where
        import H(T(T),f)
        x = f T :: T
