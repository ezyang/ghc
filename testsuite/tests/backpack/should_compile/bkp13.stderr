[1 of 2] Processing h
/--- Shape for h
provides:
requires: H -> h(hole:H):H
               T{T}
signature hole:H
\---
  [1 of 1] Compiling H[abstract sig]  ( h/H.hsig, nothing )
[2 of 2] Processing p
/--- Shape for p
provides: B -> p(hole:H):B
               T{T, T}
          A -> p(hole:H):A
               p(hole:H):B.T{p(hole:H):B.T}
requires: H -> h(hole:H):H, p(hole:H):H
               p(hole:H):B.T{p(hole:H):B.T, p(hole:H):B.T}, f
include h(hole:H) [] requires [(H, [h(hole:H):H])]
module p(hole:H):B
signature hole:H
module p(hole:H):A
\---
  [1 of 4] Including h
  [2 of 4] Compiling B                ( p/B.hs, nothing )
  [3 of 4] Compiling H[abstract sig]  ( p/H.hsig, nothing )
  [4 of 4] Compiling A                ( p/A.hs, nothing )
